### PR TITLE
Time entry report filter new design

### PR DESCRIPTION
**Pull request by onkar-saeloun**
Notion-
https://www.notion.so/saeloun/Time-Entry-Report-Desktop-Filter-Panel-design-update-c0a9d78c62584db496b861ad17c7d9d1?pvs=4

What-

1. Added new design with common components
2. Added new custom date select component
3. Added desktop view of filter panel with matching requirement

Why-

1. Common components will be reusable for other filters
2. new custom date select will be used as per new designs

Notes-

1. The logic of the custom date filter is not updated.
2. Tested on large and extra large devices for design and is working fine

Loom-
https://www.loom.com/share/14c7622dd9aa41119b64d714a3807edf

### CODE REVIEW

**Review by AkashKale — COMMENTED** [review]
- We have to remove 'All' option from Date Range
- Datepicker should close after user selects 'To' date

**Comment by onkar-saeloun** [comment]
> * We have to remove 'All' option from Date Range
> * Datepicker should close after user selects 'To' date

@AkashKale I have removed all filter but can you please have a look at the closing calendar after selecting to filter? I have created a loom video. Previously we had done and reset button on the calendar thus it was readable.
https://www.loom.com/share/c72980ada1e8418b86f2ef060f505b7f

**Review by AkashKale — COMMENTED** [review]
After the user selects from and to dates and the datepicker is closed, the textboxes should be visible to show the dates which the user has selected so that the user can change the to or from date if they want
image

**Comment by onkar-saeloun** [comment]
@AkashKale please have a look on custom date picker functionality
https://www.loom.com/share/16bbffd5b5954ac691e02957edb7d6fd

**Review by AkashKale — APPROVED** [review]

File: app/javascript/src/common/CustomDateRangeWIthInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+import {
+  validDateFormats,
+  months,
+  CUSTOM_DATE_RANGE_ERRORS,
+} from "../CustomDateRangePicker/utils";
+
+interface DateRange {
```
Comment (prasanthchaduvula) [inline app/javascript/src/common/CustomDateRangeWIthInput/index.tsx:16]: Please move the type props to the end of the file. It improves readability of the component.

File: app/javascript/src/components/Reports/Filters/FilterSideBar.tsx [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+      const sortedClients = filteredClientList.sort((a, b) =>
+        a.label.localeCompare(b.label)
+      );
+      if (debouncedSearchQuery && filteredClientList.length > 0) {
+        const newClientList = sortedClients.filter(
+          client =>
+            client.label
+              .toLowerCase()
+              .includes(debouncedSearchQuery.toLowerCase())
+          // client.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/Filters/FilterSideBar.tsx:106]: If is it not used please remove this commented code

File: app/javascript/src/components/Reports/Header/index.tsx [hunk 3]
```diff
@@ -106,14 +106,38 @@ const Header = ({
...
+                  onClick={() => {
+                    setShowExportOptions(false);
+                    handleDownload("pdf");
+                  }}
+                >
+                  <FilePdfIcon color="#5B34EA" size={16} weight="bold" />
+                  <span className="ml-3 text-sm">Export as PDF</span>
+                </button>
+              </li>
               {/* <li className="flex items-center p-2 text-miru-dark-purple-400">
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Reports/Header/index.tsx:141]: Please remove this commented code if it is not used

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/common/CustomDateRangeWIthInput/index.tsx (hunk 1 above)
Comment (onkar-saeloun) [reply app/javascript/src/common/CustomDateRangeWIthInput/index.tsx:17]: Done

File: app/javascript/src/components/Reports/Filters/FilterSideBar.tsx (hunk 2 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Reports/Filters/FilterSideBar.tsx]: Done

File: app/javascript/src/components/Reports/Header/index.tsx (hunk 3 above)
Comment (onkar-saeloun) [reply app/javascript/src/components/Reports/Header/index.tsx]: Done

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**